### PR TITLE
Fix: properly scope variables in TSModuleBlock

### LINKED ISSUE #13112: [BUG/TypeScript] `declare global` conflict with block declaration
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**

Code

```ts
declare global { namespace globalThis { var i18n: any; } }

export class i18n {}
```

trigger an scope error:

```txt
    TypeError: Duplicate declaration "i18n"
      1 | declare global { namespace globalThis { var i18n: any; } }
      2 |
    > 3 | export class i18n {}
        |              ^^^^

      249 |     }
      250 | 
    > 251 |     return new Error(msg);
          |            ^
      252 |   }
      253 | 
      254 | }

      at File.buildCodeFrameError (packages/babel-core/lib/transformation/file/file.js:251:12)
      at Scope.checkBlockScopedCollisions (packages/babel-traverse/lib/scope/index.js:422:22)
      at Scope.registerBinding (packages/babel-traverse/lib/scope/index.js:582:16)
      at Scope.registerDeclaration (packages/babel-traverse/lib/scope/index.js:527:12)
      at Object.BlockScoped (packages/babel-traverse/lib/scope/index.js:250:12)
      at Object.newFn (packages/babel-traverse/lib/visitors.js:216:17)
      at NodePath._call (packages/babel-traverse/lib/path/context.js:55:20)
      at NodePath.call (packages/babel-traverse/lib/path/context.js:38:14)
      at NodePath.visit (packages/babel-traverse/lib/path/context.js:90:31)
      at TraversalContext.visitQueue (packages/babel-traverse/lib/context.js:116:16)
```

- [REPL](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=CYUwxgNghgTiAEBzCB7ARlC8De8B2UAtiAM4AOUYCy6mAKgBYCWJO8AbrPEwIwAceAFzwoeAJ4BueAF8ZAKDkgAHmRQwALvEhQSrXgJzSgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.13.14&externalPlugins=%40babel%2Fplugin-transform-modules-systemjs%407.13.8%2C%40babel%2Fplugin-syntax-top-level-await%407.12.13)

**Input Code**

As described above.

**Expected behavior**

This piece of code is [valid](https://www.typescriptlang.org/play?ssl=3&ssc=21&pln=1&pc=1#code/CYUwxgNghgTiAEBzCB7ARlC8De8B2UAtiAM4AOUYCy6mAKgBYCWJO8AbrPEwIwAceAFzwoeAJ4BueAF8ZAKDkgAHmRQwALvEhQSrXgJzSgA) in TypeScript.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**

- Filename: `babel.config.js`

```js
{
  "presets": ["@babel/preset-typescript"]
}
```

**Environment**

- Babel version(s): [e.g. v7.12.0]
- Node/npm version: [e.g. Node 12/npm 7]
- OS: [e.g. macOS 10.15.4, Windows 10]
- Monorepo: [e.g. yes/no/Lerna]
- How you are using Babel: [e.g. `webpack`, `rollup`, `parcel`, `babel-register`]

**Possible Solution**

**Additional context**

### PR DESCRIPTION
**Pull request by The-x-Theorist**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13112  
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

scope binding will be skiped for var declarators inside TSModuleBlock.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14109"></a>

### CODE REVIEW

**Review by nicolo-ribaudo — COMMENTED** [review]
Sorry if no one reviewed this PR earlier :grimacing: 

Rather than not registering `var i18n` if it's contained inside a `declare` block, can we just avoid traversing `declare` blocks when looking for bindings?

**Comment by The-x-Theorist** [comment]
How can we avoid traversing `declare` blocks?

**Comment by nicolo-ribaudo** [comment]
You can add a `TSModuleDeclaration` function to `collectorVisitor`, and call `path.skip()` on it. Maybe it needs to be called only if `path.node.global` is `true` to fix the issue, we would need to check how it should behave with non-global module blocks.

**Comment by The-x-Theorist** [comment]
Can we just avoid adding a check for `path.node.global`, It will work with both global and non global modulel blocks?

File: packages/babel-plugin-transform-react-jsx-development/test/fixtures/linux/within-ts-module-block/output.mjs [hunk 1]
```diff
@@ -7,5 +7,5 @@ export namespace Namespaced {
     fileName: _jsxFileName,
     lineNumber: 3,
     columnNumber: 3
-  }, void 0);
+  }, this);
```
Comment (nicolo-ribaudo) [inline packages/babel-plugin-transform-react-jsx-development/test/fixtures/linux/within-ts-module-block/output.mjs:10]: tsc produces the same output, with `this`

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by liuxingbaoyu — APPROVED** [review]

**Pull request merged**